### PR TITLE
Fix the index monitor checkpoint tracking for when the documents are deleted

### PR DESCRIPTION
**Pull request by aleksmaus**
## What is the problem this PR solves?

This fixes the bug with the fleet server actions monitor that doesn't account correctly on the document sequence number increment when the expired action is deleted. This leads to the monitor checkpoint getting out of sync (lagging behind) with the index checkpoint, and causes the monitor to go into the tight loop of awaiting for checkpoint advance (which returns immediately and fetching the actions documents.

"ESS deployment being flooded by requests to /.fleet-actions/_fleet/_fleet_search. 1.3M requests over last 2 hours when customer workload is mainly Logstash indexing."

The current workaround for this would be either restart the fleet-server or push the new action, like change the logging level for example.

## How does this PR solve the problem?

Update monitor checkpoint value from the "wait advance" response upon the last fetch. This ensures that the checkpoint stays in sync with the index checkpoint even if the checkpoint advance happened due to the documents deleted from the index.

## How to test this PR locally

1. Run the stack with the fleet server and the agent.
2. Place some actions, like for example changing the agent logging level, or osquery live query.
3. Delete the action document.
4. Check the fleet server logs. The fleet server should not go into the tight loop after the deletion.

## Design Checklist

- [x] I have ensured my design is stateless and will work when multiple fleet-server instances are behind a load balancer.
- [ ] I have or intend to scale test my changes, ensuring it will work reliably with 100K+ agents connected.
- [ ] I have included fail safe mechanisms to limit the load on fleet-server: rate limiting, circuit breakers, caching, load shedding, etc. 

## Checklist

- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] I have made corresponding change to the default configuration files
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have added an entry in `CHANGELOG.next.asciidoc` or `CHANGELOG-developer.next.asciidoc`.

## Related issues

- Closes https://github.com/elastic/fleet-server/issues/2205

### CODE REVIEW

**Comment by aleksmaus** [comment]
hmmm
```
[2022-12-22T16:11:17.761Z] error obtaining VCS status: exit status 128
[2022-12-22T16:11:17.761Z] 	Use -buildvcs=false to disable VCS stamping.
[2022-12-22T16:11:17.761Z] make: *** [Makefile:150: release-darwin/amd64] Error 1
[2022-12-22T16:11:18.865Z] Makefile:173: recipe for target 'docker-release' failed
[2022-12-22T16:11:18.865Z] make: *** [docker-release] Error 2
```

File: internal/pkg/monitor/monitor_integration_test.go [hunk 1]
```diff
@@ -9,16 +9,22 @@ package monitor
 
 import (
 	"context"
-	"errors"
 	"sync"
 	"testing"
+	"time"
 
-	"github.com/google/go-cmp/cmp"
+	"github.com/rs/zerolog/log"
```
Comment (michel-laterman) [inline internal/pkg/monitor/monitor_integration_test.go:16]: nit: we have https://github.com/elastic/fleet-server/blob/main/internal/pkg/testing/log/log.go to redirect zerolog from what's tested to the `t.Log` stream

Comment (aleksmaus) [reply internal/pkg/monitor/monitor_integration_test.go]: Updated

**Comment by aleksmaus** [comment]
Tested end to end. Here are the steps performed to test this feature.

Used the local environment running 8.7.0-SNAPSHOT Elasticsearch and Kibana.

Running locally build fleet-server is a bit tricky: I downloaded the latest elastic-agent 8.7.0-SNAPSHOT docker image, run the image, copied the locally built the fleet-server binary into the container, created the new image from the container, used run that against the local elasticsearch&kibana.

Then enrolled the regular agent with osquery enabled, placed few queries, made sure the .fleet-actions documents are created and the checkpoint is incremented and tracked by the fleet-server normally.

Screen Shot 2023-01-03 at 8 48 51 AM

Screen Shot 2023-01-03 at 8 49 03 AM

Then deleted the last action, used curl with the Authroization header with the fleet server token, the elastic user doesn't have permissions to delete from .fleet-actions index.

After delete observed that:
1. The checkpoint is incremented and has value 3 in this case

Screen Shot 2023-01-03 at 8 56 55 AM

2. While the last .fleet-actions document now with the _seq_no of 1

3. (IMPORTANT!) The fleet server continues to run and the .fleet-actions monitor doesn't go into the tight loop of checking the checkpoint and fetching 0 documents. This is confirmed from the logs.

Now issue another live query. 
1. The new action document is added normally
2. The fleet-server continues to track the checkpoint (monitor works) and dispatches it to the agent
3. The query result is sent back normally 

Screen Shot 2023-01-03 at 8 57 51 AM

4. The global checkpoint is incremented as expected along with the new action document added in the index
Screen Shot 2023-01-03 at 8 58 28 AM

Screen Shot 2023-01-03 at 8 58 19 AM

**Comment by aleksmaus** [comment]
Tested end to end and regression tested this actions monitor change. Should be good to merge upon approval.

**Review by cmacknz — APPROVED** [review]
Looks good to me, with the disclaimer that I did not manually test this myself. I appreciated all the comments it made it easy to follow.

It would be nice to see this used in our scale tests before release, but it seems low risk considering that this situation isn't that common and the primary code path is unmodified.

You should likely add a change log entry for this as well.

**Pull request merged**